packages/CopyInput/src/CopyInput.js

```diff
@@ -46,12 +48,22 @@ function CopyInput(props) {
   }
 
   return (
-    <sc.CopyInput data-pka-anchor="copy-input" {...moreProps}>
-      <Input ref={inputRef} defaultValue={value} isReadOnly={isReadOnly} {...extendedInputProps} />
+    <sc.CopyInput
+      data-pka-anchor="copy-input"
+      hasDefaultButtonBorder={hasValueContainer || !hasInputContainer}
```

Comment:
I think I understand what you're trying to do here, but I was thinking we can keep it even simpler like:

```jsx
<sc.CopyInput data-pka-anchor="copy-input" hasDefaultButtonBorder={!hasInputContainer} {...moreProps}>
  {hasValueContainer ? <sc.Value>{value}</sc.Value> : null}
  {!hasInputContainer ? (
    <sc.HiddenInput ref={inputRef} defaultValue={value} />
  ) : (
    <Input isReadOnly={isReadOnly} ref={inputRef} defaultValue={value} {...extendedInputProps} />
  )}
  ...
</sc.CopyInput>
```

Which does mean that an example like this:
```jsx
<CopyInput value="something" hasValueContainer />
```

Is going to render as:

<img width="293" alt="Screen Shot 2021-06-30 at 4 59 13 PM" src="https://user-images.githubusercontent.com/14944896/124045469-8dc13900-d9c4-11eb-8458-cf6b74edf93c.png">

But 🤷‍♂️ I think that's okay, and it will be pretty easy for a consumer to understand what's happening, and how to fix it.